activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskIntegrationResultEventHandler.java

```diff
@@ -125,7 +132,41 @@ public void receive(IntegrationResult integrationResult) {
                 )
             );
 
-            managementService.executeCommand(CompositeCommand.of(commands.toArray(Command[]::new)));
+            try {
+                managementService.executeCommand(CompositeCommand.of(commands.toArray(Command[]::new)));
+            } catch (ActivitiOptimisticLockingException e) {
+                throw e;
+            } catch (Exception triggerException) {
+                LOGGER.warn(
+                    "Failed to update integration context {}. It might have been already deleted.",
+                    integrationContext.getId(),
+                    triggerException
+                );
+                IntegrationRequest fakeRequest = new IntegrationRequestImpl(integrationContext);
+                IntegrationErrorImpl integrationError = new IntegrationErrorImpl(fakeRequest, triggerException);
+                getSelf().handlePropagationFailure(integrationError, integrationContextEntity);
+            }
         }
     }
+
+    @Transactional(propagation = REQUIRES_NEW)
+    public void handlePropagationFailure(
+        IntegrationErrorImpl integrationError,
+        IntegrationContextEntity integrationContextEntity
+    ) {
+        Command<?> finalErrorHandlingCmd = CompositeCommand.of(
+            new AggregateIntegrationErrorReceivedEventCmd(
+                integrationError,
+                runtimeBundleProperties,
+                processEngineEventsAggregator
+            ),
+            new DeleteIntegrationContextCmd(integrationContextEntity)
+        );
+
+        managementService.executeCommand(finalErrorHandlingCmd);
+    }
+
+    private ServiceTaskIntegrationResultEventHandler getSelf() {
+        return applicationContext.getBean(ServiceTaskIntegrationResultEventHandler.class);
```

Comment:
Have you tried extracting `handlePropagationFailure` into a separate bean so that you don't need to use `getSelf()`?